heidrun/esdn_mods.rb

```diff
@@ -0,0 +1,80 @@
+Krikri::Mapper.define(:esdn_mods, :parser => Krikri::ModsParser) do
+  provider :class => DPLA::MAP::Agent do
+    uri 'http://dp.la/api/contributor/esdn'
+    label 'Empire State Digital Network'
+  end
+
+  dataProvider :class => DPLA::MAP::Agent do
+    providedLabel record.field('mods:note').match_attribute(:type, 'ownership')
+  end
+
+  isShownAt :class => DPLA::MAP::WebResource do
+    uri record.field('mods:location', 'mods:url').match_attribute(:usage, 'primary display').match_attribute(:access, 'object in context')
+  end
+
+  preview :class => DPLA::MAP::WebResource do
+    uri record.field('mods:location', 'mods:url').match_attribute(:access, 'preview')
+  end
+
+  originalRecord :class => DPLA::MAP::WebResource do
+    uri record_uri
+  end
+
+  sourceResource :class => DPLA::MAP::SourceResource do
+    ##
+    # TODO: Crosswalk says to take collection from OAI set name/description,
+    # but we need to be able harvest set titles and populate them somewhere.
+    # This will just pull back the setSpec code for now.
+    collection :class => DPLA::MAP::Collection, :each => header.field('xmlns:set_spec'), :as => :coll do
```

Comment:
Just noting for the record - their OAI XML refers to this element as `<set_spec/>`, but the OAI spec calls it `<setSpec/>`